### PR TITLE
Fix php8.1 compatibility for unit tests

### PR DESCRIPTION
**Pull request by SilinMykola**
### Description (*)
I changed unit tests for PHP 8.1 compatibility

### Related Issues (if relevant)
1. Fixes magento/magento2#34441

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by andrewbess — APPROVED** [review]

**Review by eduard13 — APPROVED** [review]

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
@magento run Unit Tests with env PHP 8.1 with edition CE without extensions magento-commerce/inventory, magento-commerce/adobe-stock-integration, magento-commerce/security-package, magento-commerce/magento2-page-builder, magento-commerce/adobe-ims

File: app/code/Magento/Customer/Model/FileProcessor.php [hunk 1]
```diff
@@ -215,7 +215,9 @@ public function saveTemporaryFile($fileId)
         );
 
         $result = $uploader->save($path);
-        unset($result['path']);
+        if (is_array($result)) {
```
Comment (xmav) [inline app/code/Magento/Customer/Model/FileProcessor.php]: Solution require discussion

Comment (andrewbess) [reply app/code/Magento/Customer/Model/FileProcessor.php]: Hello @xmav 
https://wiki.php.net/rfc/autovivification_false - this is reason of this solution for PHP 8.1.
The system fails if $result = false without this checking.
Also, pay attention, `$uploader->save($path)` can return boolean value

Comment (SilinMykola) [reply app/code/Magento/Customer/Model/FileProcessor.php]: I changed the solution, please check.

**Comment by xmav** [comment]
@magento run all tests

**Comment by xmav** [comment]
@magento run Functional Tests EE, Static Tests

**Comment by xmav** [comment]
Failures are not related to changes in this PR. Tests will be rechecked during delivery to 2.4-develop

**Comment by xmav** [comment]
@magento import code to magento-trigger/magento2ce

**Comment by m2-github-services** [comment]
@xmav the branch with the code is successfully imported into the `magento-trigger/magento2ce` repository. Branch name: `imported-SilinMykola-magento2-34446`.

**Pull request merged**